lanweatherd/src/sensor_loop.cpp

```diff
@@ -0,0 +1,23 @@
+#include <unistd.h>
+#include <zmq.h>
+
+#include "sensor_loop.h"
+
+int SENSOR_SLEEP_INTERVAL = 60 * 1; // every 1 minute
+
+void sensor_loop(recent_data& cache){
+    void *context = zmq_ctx_new();
+    void *responder = zmq_socket(context, ZMQ_REP);
+    zmq_bind(responder, "tcp://*:5675");
+
+    while(1){
+        char buffer[64];
+        zmq_recv(responder, buffer, 63, 0);
+        cache.update_sensor_data(buffer);
+
+        sleep(SENSOR_SLEEP_INTERVAL);
```

Comment:
`zmq_recv()` blocks, you don't need to sleep here; instead you should have the pub side of things determine the determine the frequency. i can see some weird queue-backup things happening because of how zmq handles messaging with it set up this way.

Reply:
Removed the sleep code in the most recent commit.